src/ssl.c

```diff
@@ -24561,13 +24561,52 @@ int  wolfSSL_sk_num(WOLF_STACK_OF(WOLFSSL_ASN1_OBJECT)* sk)
 
 void* wolfSSL_sk_value(WOLF_STACK_OF(WOLFSSL_ASN1_OBJECT)* sk, int i)
 {
+    #if defined(OPENSSL_ALL) || defined(WOLFSSL_QT)
+    int offset = i;
+    WOLFSSL_GENERAL_NAME* gn;
+    #endif
+    WOLFSSL_ENTER("wolfSSL_sk_value");
+
     for (; sk != NULL && i > 0; i--)
         sk = sk->next;
     if (sk == NULL)
         return NULL;
-    return (void*)sk->data.obj;
-}
 
+    switch (sk->type) {
+    #if defined(OPENSSL_ALL) || defined(WOLFSSL_QT)
```

Comment:
Some of these are available in OPENSSL_EXTRA without OPENSSL_ALL, only sk->data.access and sk->data.ext are limited to the OPENSSL_ALL/WOLFSSL_QT build.

Reply:
Check

Reply:
Also `sk->data.cipher->offset` and `WOLFSSL_GENERAL_NAME` are conditioned by OPENSSL_ALL/WOLFSSL_QT.